connect/runtime/src/test/java/org/apache/kafka/connect/integration/InternalTopicsIntegrationTest.java

```diff
@@ -0,0 +1,187 @@
+ */
+package org.apache.kafka.connect.integration;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.kafka.connect.runtime.distributed.DistributedConfig;
+import org.apache.kafka.connect.util.clusters.EmbeddedConnectCluster;
+import org.apache.kafka.test.IntegrationTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Integration test for the creation of internal topics.
+ */
+@Category(IntegrationTest.class)
+public class InternalTopicsIntegrationTest {
+
+    private static final Logger log = LoggerFactory.getLogger(InternalTopicsIntegrationTest.class);
+
+    private EmbeddedConnectCluster.Builder connectBuilder;
+    private EmbeddedConnectCluster connect;
+    Map<String, String> workerProps = new HashMap<>();
+    Properties brokerProps = new Properties();
+
+    @Before
+    public void setup() {
+        // setup Kafka broker properties
+        brokerProps.put("auto.create.topics.enable", String.valueOf(false));
+
+        // build a Connect cluster backed by Kafka and Zk
+        connectBuilder = new EmbeddedConnectCluster.Builder()
+                .name("connect-cluster")
+                .numWorkers(1)
+                .numBrokers(1)
+                .brokerProps(brokerProps);
+    }
+
+    @After
+    public void close() {
+        // stop all Connect, Kafka and Zk threads.
+        connect.stop();
+    }
+
+    @Test
+    public void testCreateInternalTopicsWithDefaultSettings() throws InterruptedException {
+        int numWorkers = 1;
+        int numBrokers = 3;
+        connect = new EmbeddedConnectCluster.Builder().name("connect-cluster-1")
+                                                      .workerProps(workerProps)
+                                                      .numWorkers(numWorkers)
+                                                      .numBrokers(numBrokers)
+                                                      .brokerProps(brokerProps)
+                                                      .build();
+
+        // Start the Connect cluster
+        connect.start();
+        connect.assertions().assertExactlyNumBrokersAreUp(numBrokers, "Brokers did not start in time.");
+        connect.assertions().assertExactlyNumWorkersAreUp(numWorkers, "Worker did not start in time.");
+        log.info("Completed startup of {} Kafka brokers and {} Connect workers", numBrokers, numWorkers);
+
+        // Check the topics
+        log.info("Verifying the internal topics for Connect");
+        connect.assertions().assertTopicsExist(configTopic(), offsetTopic(), statusTopic());
+        assertInternalTopicSettings();
+
+        // Remove the Connect worker
+        log.info("Stopping the Connect worker");
+        connect.removeWorker();
+
+        // Sleep for a bit
+        Thread.sleep(3000);
```

Comment:
did you notice that this matters? Not suggesting to remove necessarily, but I wonder whether you noticed it didn't work otherwise.

Reply:
Good catch. I had put this in while debugging an issue, but it is not needed.